src/main/java/in/projecteka/consentmanager/user/SignUpRequestValidator.java

```diff
@@ -37,16 +41,55 @@ private SignUpRequestValidator() {
                 validate(signUpRequest.getGender()),
                 validateUserName(signUpRequest.getUsername(), userIdSuffix),
                 validatePassword(signUpRequest.getPassword()),
-                validateYearOfBirth(signUpRequest.getYearOfBirth()))
-                .ap((firstName, gender, username, password, dateOfBirth) -> SignUpRequest.builder()
+                validateYearOfBirth(signUpRequest.getYearOfBirth()),
+                validateUnVerifiedIdentifiers(signUpRequest.getUnverifiedIdentifiers()))
+                .ap((firstName, gender, username, password, dateOfBirth, unverifiedIdentifiers) -> SignUpRequest.builder()
                         .name(firstName)
                         .gender(gender)
                         .username(username)
                         .password(password)
                         .yearOfBirth(dateOfBirth)
+                        .unverifiedIdentifiers(unverifiedIdentifiers)
                         .build());
     }
 
+    protected static Validation<String,List<SignUpIdentifier>> validateUnVerifiedIdentifiers(List<SignUpIdentifier> unverifiedIdentifiers) {
+        if (unverifiedIdentifiers==null || unverifiedIdentifiers.isEmpty()) {
+            return Validation.valid(unverifiedIdentifiers);
+        }
+
+        if (isInValidIdentifier(unverifiedIdentifiers)) {
+            return Validation.invalid("Invalid identifier");
+        }
+
+        boolean uniqueOrNonExistent = isUniqueOrNonExistent(unverifiedIdentifiers, IdentifierType.ABPMJAYID);
+        if (!uniqueOrNonExistent) {
+            return Validation.invalid("More than one AB-PMJAYID are not supported");
+        }
+        Optional<Boolean> optionalValidAbId = unverifiedIdentifiers.stream()
+                .filter(identifier -> identifier.getType().equals(IdentifierType.ABPMJAYID.name()))
+                .map(abIdentifier -> IdentifierType.ABPMJAYID.isValid(abIdentifier.getValue()))
+                .findFirst();
+
+        if (optionalValidAbId.isPresent() && (! optionalValidAbId.get())) {
+            return Validation.invalid("Invalid AB-PMJAYID");
+        }
+
+        return Validation.valid(unverifiedIdentifiers);
+    }
+
+    private static boolean isInValidIdentifier(List<SignUpIdentifier> unverifiedIdentifiers) {
+        return unverifiedIdentifiers.stream()
+                .anyMatch(signUpIdentifier -> Arrays.stream(IdentifierType.values())
+                        .noneMatch(identifierType -> identifierType.name().equals(signUpIdentifier.getType())));
+    }
+
+    private static boolean isUniqueOrNonExistent(List<SignUpIdentifier> unverifiedIdentifiers, IdentifierType identifierType) {
+        return unverifiedIdentifiers.stream().
+                filter(identifier -> identifier.getType().equals(identifierType.name()))
+                .count() <= 1;
+    }
```

Comment:
since it's optional, we should check uniqueness when it exists. Sorry my comment was misleading earlier.